xcel_itron2mqtt/main.py

```diff
@@ -77,5 +81,8 @@ def mDNS_search_for_meter() -> str | int:
         ip_address, port_num = mDNS_search_for_meter()
     creds = look_for_creds()
     meter = xcelMeter(INTEGRATION_NAME, ip_address, port_num, creds)
-    # The run method controls all the looping, querying, and mqtt sending
-    meter.run()
\ No newline at end of file
+    meter.setup()
+
+    if meter.initalized:
+        # The run method controls all the looping, querying, and mqtt sending
```

Comment:
So my question here. Do we want connections to be “tenacious” or should we revert to the previous step in case of an error. So first stage should be mDNS discovery, then probing for the endpoints, then the operational state. If one fails, should we revert to the previous step and only move forward if the requirements are met to proceed? I think this would make the overall service much more robust and almost never fail.

Reply:
I would agree, implementing something akin to a state machine as things start up would be the next step. I did this mostly to separate out the HASS registration from the constructor